packages/@lwc/engine-server/src/renderer.ts

```diff
@@ -377,46 +413,10 @@ const getLastChild = unsupportedMethod('getLastChild') as (element: HostElement)
 const getLastElementChild = unsupportedMethod('getLastElementChild') as (
     element: HostElement
 ) => HostElement | null;
-
-function getTagName(elm: HostElement): string {
-    // tagName is lowercased on the server, but to align with DOM APIs, we always return uppercase
-    return elm.tagName.toUpperCase();
-}
-
-/* noop */
-const assertInstanceOfHTMLElement = noop as (elm: any, msg: string) => void;
-
-type CreateElementAndUpgrade = (upgradeCallback: LifecycleCallback) => HostElement;
-
-const localRegistryRecord: Map<string, CreateElementAndUpgrade> = new Map();
-
-function createUpgradableElementConstructor(tagName: string): CreateElementAndUpgrade {
-    return function Ctor(upgradeCallback: LifecycleCallback) {
-        const elm = createElement(tagName);
-        if (isFunction(upgradeCallback)) {
-            upgradeCallback(elm); // nothing to do with the result for now
-        }
-        return elm;
-    };
-}
-
-function getUpgradableElement(tagName: string): CreateElementAndUpgrade {
-    let ctor = localRegistryRecord.get(tagName);
-    if (!isUndefined(ctor)) {
-        return ctor;
-    }
-
-    ctor = createUpgradableElementConstructor(tagName);
-    localRegistryRecord.set(tagName, ctor);
-    return ctor;
-}
-
-function createCustomElement(tagName: string, upgradeCallback: LifecycleCallback): HostElement {
-    const UpgradableConstructor = getUpgradableElement(tagName);
-    return new (UpgradableConstructor as any)(upgradeCallback);
-}
-
 const ownerDocument = unsupportedMethod('ownerDocument') as (element: HostElement) => Document;
+const attachInternals = unsupportedMethod('attachInternals') as (
+    elm: HTMLElement
+) => ElementInternals;
```

Comment:
@nolanlawson made a slight change from the [RFC](https://github.com/salesforce/lwc-rfcs/blob/nolan/element-internals/text/0000-element-internals.md?plain=1#L110) and throwing an error instead.